SocketRocket/SRWebSocket.m

```diff
@@ -427,33 +438,39 @@ - (void)_readHTTPHeader;
 
 - (void)didConnect;
 {
-    SRDebugLog(@"Connected");
+    // Potentially called through multiple code paths.
+    // Make sure we only respond once.
+    @synchronized(self) {
```

Comment:
You can avoid re-structuring all of this by simply adding `_requestRequiresSecurity` boolean variable and assigning it to `SRURLRequiresSSL()` to it.